pkg/event/manager.go

```diff
@@ -230,6 +235,82 @@ func postHookEvent(event *api.Event) {
 	log.Info("posthook of event")
 	log.Infof("worker： %s", event.Project.Worker)
 	terminateEventWorker(event)
+
+	pipeline := event.Pipeline
+	record := event.PipelineRecord
+
+	// post hooks
+	if pipeline != nil && pipeline.Notification != nil {
+
+		policy := pipeline.Notification.Policy
+
+		sendFlag := false
+		switch record.Status {
```

Comment:
If `policy` is `always`, there is no need check record status.
So we can first check `policy == "always"`, then check `record.Status` only if `policy != "always"`